Verify action added for simple account verification

Phillaf opened the pull request:
ref https://github.com/FriendsOfCake/crud-users/pull/13#issuecomment-220793203

ADmad commented on src/Action/VerifyAction.php:
```diff
@@ -0,0 +1,142 @@
+            'success' => true,
+            'entity' => $this->_table()->newEntity(),
+            'token' => $token
+        ]);
```
The subject instance is never used and is redundant.

Phillaf commented:
@admad you're right, I removed the subject from the _get method. I also added a default field for 'verified' in patchEntity, which is set to true when a user is verified.

ADmad commented on src/Action/VerifyAction.php:
```diff
@@ -0,0 +1,133 @@
+    protected function _get($token = null)
+    {
+        $token = $this->_token($token);
+        $entity = $this->_verify($this->_token($token));
```
`_token()` is already called above. Just pass `$token` to `_verify()`.

ADmad commented on src/Action/VerifyAction.php:
```diff
@@ -0,0 +1,133 @@
+    {
+        $entity = $this->_table()->patchEntity(
+            $entity,
+            [$this->_table()->aliasField('verified') => true],
```
Data to be patched shouldn't be aliased.

jippi commented on src/Action/VerifyAction.php:
```diff
@@ -0,0 +1,133 @@
+        $this->_trigger('afterVerify', $subject);
+        $this->setFlash('success', $subject);
+
+        if ($this->config('redirectUrl') === null) {
```
why not use https://github.com/FriendsOfCake/crud/blob/master/src/Traits/RedirectTrait.php ?

Phillaf replied:
See @ADmad's comment on this other pull request https://github.com/FriendsOfCake/crud-users/pull/11#discussion-diff-60635031

jippi replied:
in this case, if you don't configure `redirectUrl` and don't use `AuthComponent` this method will be sad.

```
$redirectUrl = $this->config('redirectUrl');

if (!$redirectUrl && $this->_controller()->components()->has('Auth')) {
    $redirectUrl = $this->_controller()->Auth->config('loginAction');
}
```

would probably be what you want?

ADmad replied:
I had forgotten about `RedirectTrait` when i made that comment. Better to use existing constructs instead of introducing new ones.

Phillaf commented:
Thanks for the review, the things you have pointed out are now fixed :)

jippi commented on src/Action/VerifyAction.php:
```diff
@@ -0,0 +1,126 @@
+        $this->_trigger('afterVerify', $subject);
+        $this->setFlash('success', $subject);
+
+        return $this->_redirect($subject, $this->_controller()->Auth->config('loginAction'));
```
this will fail if `AuthComponent` is not loaded?

Phillaf replied:
Yes, sorry, I got a little confused. It's patched with your suggested edit now.

jippi commented on src/Action/VerifyAction.php:
```diff
@@ -0,0 +1,133 @@
+                $redirectUrl = $this->_controller()->Auth->config('loginAction');
+        }
+
+        return $this->_redirect($subject, $this->_controller()->Auth->config('loginAction'));
```
still wrong.. should be `$redirectUrl` here

Phillaf replied:
http://i.imgur.com/k6QxdiZ.jpg 
fixed.

jippi commented:
LGTM
